sql/core/benchmarks/JoinBenchmark-results.txt

```diff
@@ -0,0 +1,80 @@
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+Join w long:                             Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+Join w long wholestage off                    4062 / 4709          5.2         193.7       1.0X
+Join w long wholestage on                      152 /  163        138.4           7.2      26.8X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+Join w long duplicated:                  Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+Join w long duplicated wholestage off         3793 / 3801          5.5         180.9       1.0X
+Join w long duplicated wholestage on           207 /  219        101.1           9.9      18.3X
+
+Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
+Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz
+
+Join w 2 ints:                           Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------
+Join w 2 ints wholestage off              138514 / 139178          0.2        6604.9       1.0X
+Join w 2 ints wholestage on               129908 / 140869          0.2        6194.5       1.1X
```

Comment:
Ur, is this correct? Previously, we had the followings.
```
     *Join w 2 ints codegen=false              4426 / 4501          4.7         211.1       1.0X
     *Join w 2 ints codegen=true                791 /  818         26.5          37.7       5.6X
```

Reply:
I think it's correct, I ran it on master:
```
build/sbt "sql/test-only *benchmark.JoinBenchmark"
......
[info] JoinBenchmark:
[info] - broadcast hash join, long key !!! IGNORED !!!
[info] - broadcast hash join, long key with duplicates !!! IGNORED !!!
Running benchmark: Join w 2 ints
  Running case: Join w 2 ints wholestage off
  Stopped after 2 iterations, 307335 ms
  Running case: Join w 2 ints wholestage on
  Stopped after 5 iterations, 687107 ms

Java HotSpot(TM) 64-Bit Server VM 1.8.0_151-b12 on Mac OS X 10.12.6
Intel(R) Core(TM) i7-7820HQ CPU @ 2.90GHz

Join w 2 ints:                           Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------
Join w 2 ints wholestage off              153532 / 153668          0.1        7321.0       1.0X
Join w 2 ints wholestage on               132075 / 137422          0.2        6297.8       1.2X
```

Reply:
Oh, interesting. Although it's beyond the scope, could you run on `branch-2.4` and `branch-2.3` please, too?